### PR TITLE
Lower ONNXDequantizeLinear to krnl

### PR DESCRIPTION
**Pull request by tungld**
This patch supports ONNXDequantizeLinear, where:
- ONNXDequantizeLinear is lowered to Krnl dialect
- Backend and LIT tests for ONNXDequantizeLinear are enables.
- Only support for per-tensor or layer dequantization. Not support for per-axis dequantization at this moment.

Besides, this patch edits `RunONNXModel.py` script to support `ui8` input.

The lowering of ONNXDequantizeLinear borrows `ONNXElementwiseUnaryOpLowering`, so this patch contains some code from #2056. It's better to land #2056 before this patch.

### CODE REVIEW

File: test/backend/inference_backend.py [hunk 1]
```diff
@@ -715,7 +718,7 @@ def get_test_models():
         "test_reduce_l2_negative_axes_keep_dims_example_cpu": {STATIC_SHAPE:{}, DYNAMIC_SHAPE:{-1:{-1}}, CONSTANT_INPUT:{-1}},
         "test_reduce_l2_negative_axes_keep_dims_random_cpu": {STATIC_SHAPE:{}, DYNAMIC_SHAPE:{-1:{-1}}, CONSTANT_INPUT:{-1}},
 
-        # ==OP== #ReduceMax
+        # ==OP== ReduceMax
```
Comment (tungld) [inline test/backend/inference_backend.py:721]: This caused `make onnx_mlir_supported_ops` failed to generate doccument files for supported ops. Fix it here.

File: src/Conversion/ONNXToKrnl/Math/Elementwise.cpp [hunk 2]
```diff
@@ -1160,15 +1200,23 @@ struct ONNXElementwiseUnaryOpLowering
       create.krnl.iterateIE(loopDef, loopDef, lbs, ubs,
           [&](KrnlBuilder &createKrnl, ValueRange loopInd) {
             Value loadedVal = createKrnl.load(X, loopInd);
+            SmallVector<Value> args;
+            args.emplace_back(loadedVal);
+            for (uint64_t i = 1; i < operands.size(); i++)
+              args.emplace_back(operands[i]);
```
Comment (tungld) [inline src/Conversion/ONNXToKrnl/Math/Elementwise.cpp:1206]: This code is borrowed from PR #2056.

File: src/Conversion/ONNXToKrnl/Math/Elementwise.cpp [hunk 3]
```diff
@@ -1160,15 +1200,23 @@ struct ONNXElementwiseUnaryOpLowering
...
+                rewriter, loc, op, elementType, args);
             // Store result in the resulting array.
             createKrnl.store(loweredOpResult, alloc, loopInd);
           });
     } else {
       Value loadedVal = create.krnl.load(X);
+      SmallVector<Value> args;
+      args.emplace_back(loadedVal);
+      for (uint64_t i = 1; i < operands.size(); i++)
+        args.emplace_back(operands[i]);
```
Comment (tungld) [inline src/Conversion/ONNXToKrnl/Math/Elementwise.cpp:1217]: This code is borrowed from PR #2056.

File: src/Conversion/ONNXToKrnl/Math/Elementwise.cpp [hunk 4]
```diff
@@ -945,6 +945,46 @@ Value emitScalarOpFor<ONNXRoundOp>(ConversionPatternRewriter &rewriter,
...
+  // y is of type f32.
+  Value XInt = scalarOperands[0];
+  Value XScale = scalarOperands[1];
+  Value XZeroPoint = scalarOperands[2];
+
+  Type xscaleTy = XScale.getType();
+  Type xzeroPointTy = XZeroPoint.getType();
+
+  // Only support scalar scale and zero_point.
+  if (!(isRankedShapedType(xscaleTy) && getRank(xscaleTy) == 0))
```
Comment (AlexandreEichenberger) [inline src/Conversion/ONNXToKrnl/Math/Elementwise.cpp:974]: NIT: this feels like an assert to me

File: utils/RunONNXModel.py [hunk 5]
```diff
@@ -296,15 +297,17 @@ def generate_random_input(input_signature, input_shapes):
         if (np.issubdtype(np_elem_type, np.floating)):
             lb = -1.0
             ub = 1.0
-
-        elif (np.issubdtype(np_elem_type, np.integer)):
-            lb = -10
+        elif (np.issubdtype(np_elem_type, np.uint8)):
```
Comment (AlexandreEichenberger) [inline utils/RunONNXModel.py:300]: NIT: is it possible to use a non-sized type (like for integers, its `np.integer`. Maybe there is a `np.unsigned`.. or if not, we could use the sizes 8, 16, 32, 64.

Comment (tungld) [reply utils/RunONNXModel.py]: I seems there is no `np.unsigned`, so I checked them case by case. Also, I added two new options `--lower-bound` and `--upper-bound` to allow users set the range for random inputs.

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM

**Pull request merged**